File: medium/EvaluateReversePolishNotation[150]/EvaluateReversePolishNotation.py
# i was having some hard time with the rust so i made a quick prototype in python
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stk = []
        for i in tokens:
            match i:
                case "+":
                    b = stk.pop()
                    a = stk.pop()
                    stk.append(a + b)
                case "-":
                    b = stk.pop()
                    a = stk.pop()
                    stk.append(a - b)
                case "/":
                    b = stk.pop()
                    a = stk.pop()
                    stk.append(int(a / b))
                case "*":
                    b = stk.pop()
                    a = stk.pop()
                    stk.append(a * b)
                case _:
                    stk.append(int(i))

        return int(stk[0])

File: medium/EvaluateReversePolishNotation[150]/EvaluateReversePolishNotation.rs
// optimal
impl Solution {
    pub fn eval_rpn(tokens: Vec<String>) -> i32 {
        let mut stk = vec![];
        for i in tokens {
            match i.as_str() {
                "+" => {
                    let b = stk.pop().unwrap();
                    let a = stk.pop().unwrap();
                    stk.push(a + b);
                }
                "-" => {
                    let b = stk.pop().unwrap();
                    let a = stk.pop().unwrap();
                    stk.push(a - b);
                }
                "*" => {
                    let b = stk.pop().unwrap();
                    let a = stk.pop().unwrap();
                    stk.push(a * b);
                }
                "/" => {
                    let b = stk.pop().unwrap();
                    let a = stk.pop().unwrap();
                    stk.push(a / b);
                }
                _ => {
                    let a = i.parse::<i32>().unwrap();
                    stk.push(a);
                }
            }
        }
        stk.pop().unwrap_or(0)
    }
}
